File: Remove Nth Node From End of List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *node = head;
        ListNode *resnode = head;
        while (n--) node = node->next;

        // n is equal to length of linked list --> remove head
        if (node == NULL) return head->next;

        while (node->next != NULL) {
        	node = node->next;
        	resnode = resnode->next;
        }
        // Skip (remove) nth node from end of list
        resnode->next = resnode->next->next;
        return head;
    }
};

File: Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
    	ListNode node = head;
    	ListNode resnode = head;
    	while (n-->0) node = node.next;
    	if (node == null) return head.next;
    	while (node.next != null) {
    		node = node.next;
            resnode = resnode.next;
    	}
        resnode.next = resnode.next.next;
        return node;
    }
}